src/evaluation/stats.jl

```diff
@@ -0,0 +1,142 @@
+using DrWatson
+using BSON
+using FileIO
+using Random
+using DataFrames
+using Statistics
+using StatsBase
+using EvalMetrics
+
+# metric names and settings 
+const BASE_METRICS = ["auc", "auprc", "tpr_5", "f1_5"]
+const PAT_METRICS = ["pat_1", "pat_5", "pat_10", "pat_20"]
+
+"""
+	_prefix_symbol(prefix, s)
+
+Modifies symbol `s` by adding `prefix` with underscore.
+"""
+_prefix_symbol(prefix, s) = Symbol("$(prefix)_$(s)")
+
+
+"""
+	_precision_at(p, scores, labels)
+
+Computes precision on portion `p` samples with highest score.
+"""
+function _precision_at(p, labels, scores)
+	pN = Int(round(floor(p*length(labels)))) # round is more forgiving
+	if pN > 0
+		sp = sortperm(scores, rev=true)[1:pN]
+		# @info sp scores[sp] labels[sp]
+		return EvalMetrics.precision(labels[sp], ones(eltype(labels), pN))
+	else
+		return NaN
+	end
+end
+
+"""
+	compute_stats(f::String)
+
+Computes evaluation metrics from the results of experiment in serialized bson at path `f`.
+Returns a DataFrame row with metrics and additional metadata for groupby's.
+Hash of the model parameters is precomputed in order to make the groupby easier.
+"""
+function compute_stats(f::String)
+	r = load(f)
+	row = (
+		modelname = r[:modelname],
+		dataset = r[:dataset],
+		phash = hash(r[:parameters]),
+		parameters = savename(r[:parameters], digits=6), 
```

Comment:
It's a good idea to compare models with the hash, since savename ignores array arguments.